src/GafferScene/Preview/RendererAlgo.cpp

```diff
@@ -512,6 +513,21 @@ struct LocationOutput
 				updatedAttributes->members()[g_setsAttributeName] = boost::const_pointer_cast<InternedStringVectorData>( setsAttribute );
 			}
 
+			// TODO: clean up
+			CompoundObject::ObjectMap::const_iterator it = updatedAttributes->members().find( "linkedLightsSet" ); // TODO: naming needs to be determined
```

Comment:
But `updatedAttributes` is initialised from `m_attributes`, which is inherited from the parent. So if you do the `find()` in `updatedAttributes` you would be redoing the work for every single descendant of the location that actually had the attribute assigned.